src/EnergyPlus/SolarShading.cc

```diff
@@ -6420,31 +6402,29 @@ namespace SolarShading {
 
                         NINSHC = LOCHCA - FINSHC + 1;
                         if (NINSHC <= 0) continue;
-                        OverlapArea = HCAREA(FINSHC);
+                        Real64 OverlapArea = HCAREA(FINSHC);
                         for (int J = 2; J <= NINSHC; ++J) {
                             OverlapArea += HCAREA(FINSHC - 1 + J) * (1.0 - HCT(FINSHC - 1 + J));
                         }
 
                         if (OverlapArea > 0.001) {
                             ++JBKS;
-                            if (Surface(BackSurfNum).BaseSurf == BackSurfNum) JBKSbase = JBKS;
                             if (JBKS <= MaxBkSurf) {
                                 BackSurfaces(TS, iHour, JBKS, HTSS) = BackSurfNum;
-                                // Remove following IF check: multiplying by sunlit fraction in the following is incorrect
-                                // (FCW, 6/28/02)
-                                // IF (WindowRevealStatus(HTSS,IHOUR,TS) == WindowShadedOnlyByReveal) THEN
-                                //  OverlapArea = OverlapArea*(SAREA(HTSS)/Surface(HTSS)%Area)
-                                // ENDIF
+                                int baseSurfaceNum = DataSurfaces::Surface(BackSurfNum).BaseSurf;
```

Comment:
Changes in this function to remove old surface order dependency (where subsurfaces were assumed to follow after their base surface).

Reply:
:+1: